compat/src/suspense.js

```diff
@@ -61,40 +53,32 @@ Suspense.prototype = new Component();
 Suspense.prototype._childDidSuspend = function(promise) {
 	/** @type {import('./internal').SuspenseComponent} */
 	const c = this;
-	c._suspensions.push(promise);
 
 	const onSuspensionComplete = () => {
-		// From https://twitter.com/Rich_Harris/status/1125850391155965952
-		c._suspensions[c._suspensions.indexOf(promise)] =
-			c._suspensions[c._suspensions.length - 1];
-		c._suspensions.pop();
-
-		if (c._suspensions.length == 0) {
-			// If fallback is null, don't try to unmount it
-			// `unmount` expects a real VNode, not null values
-			if (c._fallback) {
-				// Unmount current children (should be fallback)
-				unmount(c._fallback);
-			}
-			c._vnode._dom = null;
-
-			c._vnode._children = c.state._parkedChildren;
-			c.setState({ _parkedChildren: null });
+		if (!--c._suspensions) {
+			c._detachOnNextRender = false;
+			c._vnode._children[0]._children = c.state._suspended;
+			c.setState({ _suspended: null });
 		}
 	};
 
-	if (c.state._parkedChildren == null) {
-		c._fallback = c._fallback && cloneElement(c._fallback);
-		c.setState({ _parkedChildren: c._vnode._children });
-		detachDom(c._vnode._children);
-		c._vnode._children = [];
+	if (!c._suspensions++) {
+		c._detachOnNextRender = true;
+		c.setState({ _suspended: c._vnode._children[0]._children });
 	}
-
 	promise.then(onSuspensionComplete, onSuspensionComplete);
 };
 
 Suspense.prototype.render = function(props, state) {
-	return state._parkedChildren ? this._fallback : props.children;
+	if (this._detachOnNextRender) {
+		detachDom(this._vnode._children[0])._children = [];
+		this._detachOnNextRender = false;
+	}
+
+	return [
+		createElement(Fragment, null, state._suspended ? null : props.children),
+		createElement(Fragment, null, state._suspended && props.fallback)
```

Comment:
I'm wondering if we can skip the inner `Fragments` here. Returning an array from `render` will create one and maybe we could just pass the suspensions as direct children.

Reply:
Not sure whether the trick is to have two never un-mounting children rendered by suspense tricking our diff algo, isn't it?

Reply:
Yes. This was the smallest variant I found that keeps the tree consistent enough for the diff algorithm.

Reply:
Oh right, and I did test this with @marvinhagemeister's suggestion but the DOM elements started moving around again :)